typescript/phoenix-sdk/src/utils/solana.ts

```diff
@@ -1,13 +1,19 @@
 import * as beet from "@metaplex-foundation/beet";
+import { Connection } from "@solana/web3.js";
 
 export type Cluster = "mainnet-beta" | "devnet" | "localhost";
 
-export function getClusterFromEndpoint(endpoint: string): Cluster {
-  if (endpoint.includes("dev")) return "devnet";
-  if (endpoint.includes("local") || endpoint.includes("127.0.0.1"))
+export async function getClusterFromConnection(
```

Comment:
Let's make sure this isn't called too frequently (needs to be cached somewhere), don't want to make a network call every time